server/error.go

```diff
@@ -1,46 +1,13 @@
 package server
 
 import (
-	kithttp "github.com/go-kit/kit/transport/http"
 	"github.com/juju/errgo"
 )
 
-func errorDomain(err error) string {
-	switch e := err.(type) {
-	case kithttp.Error:
-		switch e.Domain {
-		case kithttp.DomainEncode:
-			return "encode"
-		case kithttp.DomainDecode:
-			return "decode"
-		case kithttp.DomainDo:
-			return "domain"
-		}
-	}
-	return "server"
-}
-
-func errorMessage(err error) string {
-	if err == nil {
-		return ""
-	}
-	switch kitErr := err.(type) {
-	case kithttp.Error:
-		switch errgoErr := kitErr.Err.(type) {
-		case *errgo.Err:
-			return errgoErr.Error()
-		}
-	}
-	return err.Error()
-}
-
 func errorTrace(err error) string {
-	switch kitErr := err.(type) {
-	case kithttp.Error:
-		switch errgoErr := kitErr.Err.(type) {
-		case *errgo.Err:
-			return errgoErr.GoString()
-		}
+	switch e := err.(type) {
+	case *errgo.Err:
+		return e.GoString()
```

Comment:
The error handling changed and was a bit weird but not that difficult. The error handling is now simplified because there is no weird error type and error differentiation in go-kit itself anymore. In the upcoming PRs in our microservices we will see how the error handling in the error encoders of the server packages changes.